### DISCUSSION TITLE
Which Speechly gRPC services are exposed as HTTP REST?

### DISCUSSION THREAD

**Discussion by rmtuckerphx** [speechly/speechly#196] [Q&A] [answered]
You can call Identity Login over HTTP:

```
$ curl https://api.speechly.com/speechly.identity.v2.IdentityAPI/Login -d '{"deviceId": "guid1", "application": {"appId": "guid2"}}'

{
 "token": "aaa.bbb.ccc,
 "validForS": 86399,
 "expiresAtEpoch": "1654018420",
 "expiresAt": "2022-05-31T17:33:40.724968+00:00"
}
```

Can the same be done for WLU/Text and SLU/Stream?

**Comment by langma** [comment, marked as answer]
The WLU API is accessible over HTTP, you just need to set the `Authorization` header properly (use the token you got in the login):
```
$ curl https://api.speechly.com/speechly.slu.v1.WLU/Text -H "Authorization: Bearer $TOKEN" -d '{"text": "show python repos"}'
{
 "segments": [
  {
   "text": "SHOW PYTHON REPOS",
   "tokens": [
    {
     "word": "SHOW",
     "index": 0
    },
    {
     "word": "PYTHON",
     "index": 1
    },
    {
     "word": "REPOS",
     "index": 2
    }
   ],
   "entities": [
    {
     "entity": "language",
     "value": "PYTHON",
     "startPosition": 1,
     "endPosition": 2
    }
   ],
   "intent": {
    "intent": "filter"
   },
   "annotatedText": "*filter show [python|python](language) repos"
  }
 ]
}
```

For the `SLU/Stream` the situation is a bit different, as it is a bidirectional stream (both client and server are streaming messages asynchronously). This cannot be achieved via HTTP in a generic fashion, so we need to use a WebSocket for that. At the moment we recommend our browser-client if you want to use the API from a browser, which will make the state and WebSocket management for you.
